src/Plotting/plot_recipes.jl

```diff
@@ -203,6 +285,15 @@ julia> plot(Singleton([0.5, 1.0]))
     seriescolor --> DEFAULT_COLOR
     seriestype := :scatter
 
+    # update fixed plot limits if necessary
+    p = plotattributes[:plot_object]
+    if length(p) > 0
+        lims = _extract_limits(p)
+        _update_plot_limits!(lims, S)
+        xlims --> lims[:x]
+        ylims --> lims[:y]
+    end
```

Comment:
It would be nice to have a macro:
```julia
macro update_plot_limits!(S)
    p = plotattributes[:plot_object]
    if length(p) > 0
        lims = _extract_limits(p)
        _update_plot_limits!(lims, S)
        xlims --> lims[:x]
        ylims --> lims[:y]
    end
end
```

Reply:
I can work on it.

Reply:
Hopefully you can just copy-paste what I wrote (maybe with some additional `quote`).

Reply:
There might be problems with a "macro call within a macro". If it does not work immediately, we can defer it to a new issue. So do not waste time on it.

Reply:
I am not sure why, but I didn't work immediately, so let's open a new issue.

Reply:
Agreed. I added a list of follow-up issues to the OP.